### PR TITLE
✨(marsha) add cronjob for management command check_harvested

### PR DESCRIPTION
**Pull request by lunika**
## Purpose

The management command check_harvested should be executed once a day.
For this we add a cronjob in marsha app configured to run the management
command every day at 3am

## Proposal

- [x] add cronjob for management command check_harvested

### CODE REVIEW

File: apps/marsha/templates/services/app/cronjob_check_harvested.yml.j2 [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  suspend: {{ suspend_cronjob | default(false) }}
+  jobTemplate:
+    spec:
+      template:
+        metadata:
+          name: "check-harvested-{{ deployment_stamp }}"
+          labels:
+            app: marsha
+            service: app
+            version: "{{ marsha_image_tag }}"
```
Comment (sampaccoud) [inline apps/marsha/templates/services/app/cronjob_check_harvested.yml.j2:25]: I think we stopped putting this label?

File: apps/marsha/templates/services/app/cronjob_check_harvested.yml.j2 [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                - name: POSTGRES_DB
+                  value: "{{ marsha_postgresql_database }}"
+                - name: POSTGRES_HOST
+                  value: "marsha-{{ marsha_postgresql_host }}-{{ deployment_stamp }}"
+                - name: POSTGRES_PORT
+                  value: "{{ marsha_postgresql_port }}"
+                - name: DJANGO_ALLOWED_HOSTS
+                  value: "{{ marsha_host | blue_green_hosts }}"
+                - name: DJANGO_CLOUDFRONT_PRIVATE_KEY_PATH
+                  value: "{{ marsha_cloudfront_private_key_path }}"
```
Comment (sampaccoud) [inline apps/marsha/templates/services/app/cronjob_check_harvested.yml.j2:55]: you don't need these 2 but ok if you want the template to the same on all jobs.

File: apps/marsha/templates/services/app/cronjob_check_harvested.yml.j2 [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+metadata:
+  labels:
+    app: marsha
+    service: app
+    version: "{{ marsha_image_tag }}"
+    deployment_stamp: "{{ deployment_stamp }}"
+  name: "check-harvested-{{ deployment_stamp }}"
+  namespace: "{{ project_name }}"
+spec:
+  schedule: "0 3 * * *"
```
Comment (madmatah) [inline apps/marsha/templates/services/app/cronjob_check_harvested.yml.j2]: Shouldn't you introduce a new variable to make this scheduling configurable?

File: CHANGELOG.md [hunk 4]
```diff
@@ -8,6 +8,10 @@ Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## Unreleased
 
+### Added
+
+- Marsha: cronjob running check_harvested management command
```
Comment (sampaccoud) [inline CHANGELOG.md:13]: Don't forget to port this PR to @madmatah's branch https://github.com/openfun/arnold/tree/migration-to-k8s to avoid a regression.

**Review by sampaccoud — APPROVED** [review]

File: apps/marsha/templates/services/app/cronjob_check_harvested.yml.j2 (hunk 3 above)
Comment (lunika) [reply apps/marsha/templates/services/app/cronjob_check_harvested.yml.j2]: Yes but I should do it in the other cronjob too

File: CHANGELOG.md (hunk 4 above)
Comment (lunika) [reply CHANGELOG.md:13]: Yes I plan to do it.

File: apps/marsha/templates/services/app/cronjob_check_harvested.yml.j2 (hunk 2 above)
Comment (lunika) [reply apps/marsha/templates/services/app/cronjob_check_harvested.yml.j2:55]: If I remember, this setting is mandatory. Running the management command will saying that this setting is missing.

File: apps/marsha/templates/services/app/cronjob_check_harvested.yml.j2 (hunk 1 above)
Comment (lunika) [reply apps/marsha/templates/services/app/cronjob_check_harvested.yml.j2:25]: Ok I didn't know. But we are still using them in marsha at least.

File: apps/marsha/templates/services/app/cronjob_check_harvested.yml.j2 (hunk 2 above)
Comment (sampaccoud) [reply apps/marsha/templates/services/app/cronjob_check_harvested.yml.j2:55]: ok then :man_shrugging:

File: apps/marsha/templates/services/app/cronjob_check_harvested.yml.j2 (hunk 1 above)
Comment (sampaccoud) [reply apps/marsha/templates/services/app/cronjob_check_harvested.yml.j2:25]: that's part of @madmatah's recent work.

Comment (madmatah) [reply apps/marsha/templates/services/app/cronjob_check_harvested.yml.j2:25]: The label `version` has been removed on `Deployment`s (in `spec.selector` section ) and in `Static Services`.
Its use has not changed for the rest of the k8s entities.

**Review by madmatah — APPROVED** [review]

**Pull request merged**